### PR TITLE
fix(ci): narrow down what branches to build

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It tells the CI to build those branches with following names:

- master
- 7.x
- 7.10+
- PRs
- releases (v.x.y.z)

## Why is it important?
To avoid building branches that are already PRs

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Screenshots
Screenshot 2021-07-26 at 14 05 20

### CODE REVIEW

File: .ci/jobs/fleet-server.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@
         discover-pr-forks-trust: permission
         discover-pr-origin: merge-current
         discover-tags: true
-        head-filter-regex: '^(?!update-.*-version).*$'
+        head-filter-regex: '(master|7\.s|7\.1\d|8\.\d+|PR-.*|v\d+\.\d+\.\d+|(?!update-.*-version).*$)'
```
Comment (mdelapenya) [inline .ci/jobs/fleet-server.yml:16]: Using adapted regex from Beats repo

Comment (mdelapenya) [inline .ci/jobs/fleet-server.yml:16]: Typo:
```suggestion
        head-filter-regex: '(master|7\.x|7\.1\d|8\.\d+|PR-.*|v\d+\.\d+\.\d+|(?!update-.*-version).*$)'
```

File: .ci/jobs/fleet-server.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@
         discover-pr-forks-trust: permission
         discover-pr-origin: merge-current
         discover-tags: true
-        head-filter-regex: '^(?!update-.*-version).*$'
+        head-filter-regex: '(master|7\.x|7\.1\d|8\.\d+|PR-.*|v\d+\.\d+\.\d+|(?!update-.*-version).*$)'
```
Comment (mdelapenya) [inline .ci/jobs/fleet-server.yml:16]: Using adapted regex from Beats repo

Comment (v1v) [reply .ci/jobs/fleet-server.yml:16]: `(?!update-.*-version)` is not needed then

```suggestion
        head-filter-regex: '(master|7\.x|7\.1\d|8\.\d+|PR-.*|v\d+\.\d+\.\d+)'
```

Comment (mdelapenya) [reply .ci/jobs/fleet-server.yml:16]: OK, I kept it in the case it was needed, but you are right: it's excluding that pattern. 👍

**Comment by mdelapenya** [comment]
Waiting for the Fleet Server team review to merge

**Review by v1v — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Review by ruflin — APPROVED** [review]
LGTM.

**Pull request merged**